TextToSpeech/impl/TTSDownloader.cpp

```diff
@@ -57,77 +57,71 @@ void TTSDownloader::download(TTSConfiguration config)
-       std::string tts_request;
-       TTSURLConstructer url;
-       m_objectMutex.lock();
-       tts_request = url.constructURL(m_config,"",true);
-       m_objectMutex.unlock();
-       while(tts_request.empty())
-       {
-          sleep(60);
-          //looks like no internet..wait for 1 min and retry
-          m_objectMutex.lock();
-          tts_request = url.constructURL(m_config,"",true);
-          m_objectMutex.unlock();
-       }
-       if((tts_request.compare("null")) == 0)
-       {
-          //got response ..invalid x-api-key set by app
-          TTSLOG_INFO("TTSDownloader::download got invalid response from server\n");
-          continue;
-       }
-       TTSLOG_INFO("TTSDownloader::download going to download file from location %s\n",tts_request.c_str());
-       downloadFile(tts_request);
+        if(active == false)
+        {
+            //invoked from destructor..no download
+            break;
+        }
+        needDownload = false;
+        std::string tts_request;
+        TTSURLConstructer url;
+        m_objectMutex.lock();
+        tts_request = url.constructURL(m_config,"",true);
+        m_objectMutex.unlock();
+        if((tts_request.compare("null")) == 0)
+        {
+            //got response ..invalid x-api-key set by app
+            TTSLOG_INFO("TTSDownloader::download got invalid response from server\n");
+            continue;
+        }
+        TTSLOG_INFO("TTSDownloader::download going to download file from location %s\n",tts_request.c_str());
+        downloadFile(tts_request);
     }
 }
 
 void TTSDownloader::downloadFile(std::string tts_request)
 {
-   CURL *curl;
-   CURLcode res;
-   curl = curl_easy_init();
-   if(curl)
-   {
-       char outfilename[FILENAME_MAX] = CONFIG_FILE;
-       FILE *fp;
-       fp = fopen(outfilename,"wb");
-       curl_easy_setopt(curl, CURLOPT_URL, tts_request.c_str());
-       curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, NULL);
-       curl_easy_setopt(curl, CURLOPT_WRITEDATA, fp);
-       res = curl_easy_perform(curl);
-       if ( res != CURLE_OK )
-       {
-          TTSLOG_INFO("TTSDownloader::download file download failed..needs re-download\n");
-          needDownload = true; //need re-download
-       }
-       else
-       {
-          saveConfiguration(CONFIG_PATH);
-       }
-       curl_easy_cleanup(curl);
-       fclose(fp);
+    CURL *curl;
+    CURLcode res;
+    curl = curl_easy_init();
+    if(curl)
+    {
+        char outfilename[FILENAME_MAX] = CONFIG_FILE;
+        FILE *fp;
+        fp = fopen(outfilename,"wb");
+        curl_easy_setopt(curl, CURLOPT_URL, tts_request.c_str());
+        curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, NULL);
+        curl_easy_setopt(curl, CURLOPT_WRITEDATA, fp);
+        res = curl_easy_perform(curl);
+        if ( res != CURLE_OK )
+        {
+            TTSLOG_INFO("TTSDownloader::downloadFile download failed..needs re-download\n");
+            needDownload = true; //need re-download
```

Comment:
Would it be more clear if we do just the downloading part here and moving sleep() & flagging needDownload to downloadThread()? This way I believe needDownload can be easily tracked as it is updated by the same function

Reply:
ok